assembly/src/parsers/field_ops.rs

```diff
@@ -78,9 +78,9 @@ pub fn parse_div(op: &Token) -> Result<Node, ParsingError> {
         0 => unreachable!(),
         1 => Ok(Instruction(Div)),
         2 => {
-            let imm = parse_element_param(op, 1)?;
-            check_div_by_zero(imm.as_int(), op, 1)?;
-            Ok(Instruction(DivImm(imm)))
+            let imm = parse_checked_param::<u64>(op, 1, 0, Felt::MODULUS)?;
```

Comment:
The upper bound should be `Felt::MODULUS - 1`. But I think in this case, it is fine to use `parse_felt()` and then use `as_int()` for `check_div_by_zero()`. It is a bit more costly, but keeps all the code in the same place.